### ISSUE TITLE
"bad credentials" when logging in with created user

### ISSUE THREAD

**Issue by benjamingeer** [dasch-swiss/knora-py#17]
Using the provided `example-onto.json`, a user `testuser` is created:

```
$ knora-create-ontology example-onto.json 
/usr/local/lib/python3.7/site-packages/knora
Ontology is syntactically correct and passed validation!
{"shortname": "kpt", "shortcode": "00FE", "longname": "Test-ontology for knora-py", "status": true, "selfjoin": false, "description": [{"language": "en", "value": "A ontology for testing knora-py"}], "keywords": ["test", "knora-py"]}
Project-IRI: http://rdfh.ch/projects/00FE
Creating lists...
Creating list:options
Adding users...
User added: testuser
Creating the ontology...
Creating the resclasses...
Created resclass: object1
Created resclass: object2
Creating the properties...
Property created: textprop (kpt:textprop)
Property created: intprop (kpt:intprop)
Property created: listprop (kpt:listprop)
Property created: dateprop (kpt:dateprop)
Property created: decimalprop (kpt:decimalprop)
Property created: geonameprop (kpt:geonameprop)
Property created: richtextprop (kpt:richtextprop)
Property created: intervalprop (kpt:intervalprop)
Property created: titleprop (kpt:titleprop)
Property created: linkprop (kpt:linkprop)
Adding cardinalities...
Cardinality for textprop added: 1
Cardinality for intprop added: 1
Cardinality for listprop added: 1
Cardinality for dateprop added: 1-n
Cardinality for decimalprop added: 0-1
Cardinality for geonameprop added: 0-1
Cardinality for richtextprop added: 0-1
Cardinality for intervalprop added: 0-1
Cardinality for titleprop added: 1
Cardinality for linkprop added: 0-n
```

But if I try to log in with that user's credentials, I get an error. Test program:

```python
#!/usr/bin/env python3

from knora import Knora

con = Knora("http://0.0.0.0:3333")
con.login("testuser@test.org", "test")
```

Running the program gives this error:

```
$ ./test.py 
Traceback (most recent call last):
  File "./test.py", line 6, in <module>
    con.login("testuser@test.org", "test")
  File "/usr/local/lib/python3.7/site-packages/knora/knora.py", line 137, in login
    self.on_api_error(req)
  File "/usr/local/lib/python3.7/site-packages/knora/knora.py", line 163, in on_api_error
    raise KnoraError("KNORA-ERROR: status code=" + str(res.status_code) + "\nMessage:" + res.text)
knora.knora.KnoraError: KNORA-ERROR: status code=401
Message:{"knora-api:error":"org.knora.webapi.BadCredentialsException: bad credentials: not valid","@context":{"knora-api":"http://api.knora.org/ontology/knora-api/v2#"}}
Exception ignored in: <function Knora.__del__ at 0x105a6f050>
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/knora/knora.py", line 153, in __del__
  File "/usr/local/lib/python3.7/site-packages/knora/knora.py", line 147, in logout
AttributeError: 'Knora' object has no attribute 'token'
```

**Comment by lrosenth** [comment]
@subotic Yes, I have the same problem. Seems to be a problem with creating the user using the admin API.

**Comment by subotic** [comment]
I will take a look as soon as I get my Python installation working on macOS Catalina. Running python triggers the crash reporter.

**Comment by subotic** [comment]
So, with the current version in the repository, all tests in `tests/test_knora.py` pass. To install the version from this repository:

1. `$ pip3 uninstall knora`
2. `$ pip3 install .` or `$pip3 install --editable .` if you want to work on `knora-py`

Then to run the tests:
`$ pytest`

The `knora-stack` needs to be running.

Maybe the test is wrong. I will work on it tomorrow.

**Comment by subotic** [comment]
I've improved the test, but it still passes.

I'm also able to login using Salsah1 with the created user. There must be a problem with `create_ontology.py`, which I'm going to check now.

**Comment by subotic** [comment]
I've found and fixed the problem (hopefully).

**Comment by subotic** [comment]
fixed in 9db64455454bdf5281ea1642eac6206aacaf092b and 3583ea2cafc45adecc1f422e21b3e4d145879e34
